### PR TITLE
Update product rest config and template block usage

### PR DESCRIPTION
**Pull request by louwie17**
### All Submissions:

-   [x] Have you followed the [WooCommerce Contributing guideline](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md)?
-   [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
-   [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change?

### Changes proposed in this Pull Request:

This PR updates the `rest_base` and `rest_namespace` of the product post type to use the WC REST api.
To keep the template working, the block editor required a couple small changes as well.

### How to test the changes in this Pull Request:

- [ ] Have you followed the [Writing high-quality testing instructions guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions)?

1. Make sure you have the `block-editor-feature-enabled` feature enabled and go to **WooCommerce > Products > Add new**
2. Make sure things load correctly
3. Now go to **WooCommerce > Products > All products** and click edit on a product, the block editor should load with the name correctly filled in.

### Other information:

-   [x] Have you added an explanation of what your changes do and why you'd like us to include them?
-   [ ] Have you written new tests for your changes, as applicable?
-   [x] Have you created a changelog file for each project being changed, ie `pnpm --filter=<project> changelog add`?
-   [x] Have you included testing instructions?

### FOR PR REVIEWER ONLY:

-   [ ] I have reviewed that everything is sanitized/escaped appropriately for any SQL or XSS injection possibilities. I made sure Linting is not ignored or disabled.

### CODE REVIEW

File: packages/js/product-editor/src/components/block-editor/block-editor.tsx [hunk 1]
```diff
@@ -95,13 +100,16 @@ export function BlockEditor( {
 
 	useLayoutEffect( () => {
 		setupEditor( product, {}, _settings?.template );
```
Comment (mdperez86) [inline packages/js/product-editor/src/components/block-editor/block-editor.tsx]: This line is breaking the editor cause the product does not contains a prop `content.raw` so we need to skip this or implement something similar.

**Review by mdperez86 — COMMENTED** [review]
@louwie17 Nice work here! I just left a comment.

File: packages/js/product-editor/src/components/block-editor/block-editor.tsx (hunk 1 above)
Comment (louwie17) [reply packages/js/product-editor/src/components/block-editor/block-editor.tsx]: Good catch!

Comment (louwie17) [reply packages/js/product-editor/src/components/block-editor/block-editor.tsx]: Thanks @mdperez86, this seemed to only be a problem for new products as it checked for the `auto-draft` status.
Although the way we were using it, we are better off using `setupEditorState` which `setupEditor` uses internally.

This should be fixed now as part of: https://github.com/woocommerce/woocommerce/pull/37206/commits/abb50e839ea898682b7ff153f4f1e36d8577d49f

**Review by mdperez86 — APPROVED** [review]
@louwie17 Nice work here! It looks good to me. Thanks.

**Pull request merged**